app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt

```diff
@@ -20,161 +20,54 @@ package org.kiwix.kiwixmobile.main
-      if (zimFile != null && File(zimFile).exists()) {
-        Log.d(
-          TAG_KIWIX,
-          "Kiwix normal start, zimFile loaded last time -> Open last used zimFile $zimFile"
-        )
-        restoreTabStates()
-        // Alternative would be to restore webView state. But more effort to implement, and actually
-        // fits better normal android behavior if after closing app ("back" button) state is not maintained.
-      } else {
-        Log.d(TAG_KIWIX, "Kiwix normal start, no zimFile loaded last time  -> display home page")
-        showHomePage()
-      }
-    }
-  }
-
   override fun injection(coreComponent: CoreComponent) {
     kiwixActivityComponent.inject(this)
   }
 
-  override fun hasValidFileAndUrl(url: String?, zimFileReader: ZimFileReader?) =
-    super.hasValidFileAndUrl(url, zimFileReader) && url != HOME_URL
-
-  override fun getIconResId() = R.mipmap.ic_launcher
-
-  override fun urlIsInvalid() =
-    super.urlIsInvalid() || currentWebView.url == HOME_URL
-
-  override fun showHomePage() {
-    currentWebView.removeAllViews()
-    currentWebView.loadUrl(HOME_URL)
-  }
-
-  override fun createNewTab() {
-    newTab(HOME_URL)
+  override fun onActionModeStarted(mode: ActionMode) {
+    super.onActionModeStarted(mode)
+    supportFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
+      it.onActionModeStarted(mode, this)
+    }
   }
 
-  override fun isInvalidTitle(zimFileTitle: String?) =
-    super.isInvalidTitle(zimFileTitle) || HOME_URL == currentWebView.url
-
-  private fun uriFromIntent() =
-    intent.data ?: intent.getStringExtra(EXTRA_ZIM_FILE)?.let {
-      File(FileUtils.getFileName(it)).toUri()
+  override fun onActionModeFinished(mode: ActionMode) {
+    super.onActionModeFinished(mode)
+    supportFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
+      it.onActionModeFinished(mode, this)
     }
+  }
 
-  private fun restoreTabStates() {
-    val settings = getSharedPreferences(PREF_KIWIX_MOBILE, 0)
-    val zimFile = settings.getString(TAG_CURRENT_FILE, null)
-    val zimArticles = settings.getString(TAG_CURRENT_ARTICLES, null)
-    val zimPositions = settings.getString(TAG_CURRENT_POSITIONS, null)
-
-    val currentTab = settings.getInt(TAG_CURRENT_TAB, 0)
-
-    if (zimFile != null) {
-      openZimFile(File(zimFile))
-    } else {
-      currentWebView.snack(R.string.zim_not_opened)
-    }
-    try {
-      val urls = JSONArray(zimArticles)
-      val positions = JSONArray(zimPositions)
-      var i = 0
-      currentWebView.loadUrl(reformatProviderUrl(urls.getString(i)))
-      currentWebView.scrollY = positions.getInt(i)
-      i++
-      while (i < urls.length()) {
-        newTab(reformatProviderUrl(urls.getString(i)))
-        currentWebView.scrollY = positions.getInt(i)
-        i++
+  override fun onBackPressed() {
+    supportFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
```

Comment:
these delegating calls are duplicated across Kiwix/CustomMainActivity they should be moved to Core